registry.go

```diff
@@ -1,20 +1,36 @@
 package distribution
 
-import (
-	"io"
-	"time"
+import "golang.org/x/net/context"
 
-	"github.com/docker/distribution/digest"
-	"github.com/docker/distribution/manifest"
-	"golang.org/x/net/context"
-)
+// Scope defines the set of items that match a namespace.
+type Scope interface {
+	// Contains returns true if the name belongs to the namespace.
+	Contains(name string) (bool, error)
```

Comment:
What's the case you would expect this to throw an error if its just doing a string comparison?